src/main/java/com/minecolonies/entity/EntityFishHook.java

```diff
@@ -748,21 +712,40 @@ private ItemStack getFishingLoot(@NotNull final EntityCitizen citizen)
 
         if (random < speedBonus || buildingLevel == 1)
         {
-            return ((WeightedRandomFishable) WeightedRandom.getRandomItem(this.rand, fishDrops)).getItemStack(this.rand);
+            LootContext.Builder lootcontext$builder = new LootContext.Builder((WorldServer) this.worldObj);
+            for (ItemStack itemstack : this.worldObj.getLootTableManager()
+                                         .getLootTableFromLocation(LootTableList.GAMEPLAY_FISHING)
+                                         .generateLootForPools(this.rand, lootcontext$builder.build()))
+            {
+                return itemstack;
+            }
         }
         else
         {
             random -= speedBonus;
 
             if (random < lootBonus || buildingLevel == 2)
             {
-                return ((WeightedRandomFishable) WeightedRandom.getRandomItem(this.rand, junkDrops)).getItemStack(this.rand);
+                LootContext.Builder lootcontext$builder = new LootContext.Builder((WorldServer) this.worldObj);
+                for (ItemStack itemstack : this.worldObj.getLootTableManager()
+                                             .getLootTableFromLocation(LootTableList.GAMEPLAY_FISHING_JUNK)
+                                             .generateLootForPools(this.rand, lootcontext$builder.build()))
+                {
+                    return itemstack;
+                }
             }
             else
             {
-                return ((WeightedRandomFishable) WeightedRandom.getRandomItem(this.rand, rareDrops)).getItemStack(this.rand);
+                LootContext.Builder lootcontext$builder = new LootContext.Builder((WorldServer) this.worldObj);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00117)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'lootcontext$builder' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)